Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -164,17 +174,16 @@ internal void OnException(Exception exception)
         /// </summary>
         internal void OnActivityStart(HttpRequestMessage request)
         {
-            if (Activity.Current == null)
+            var currentActivity = Activity.Current;
+            if (currentActivity == null)
             {
                 DependencyCollectorEventSource.Log.CurrentActivityIsNull();
                 return;
             }
 
-            if (request != null &&
-                !this.applicationInsightsUrlFilter.IsApplicationInsightsUrl(request.RequestUri))
```

Comment:
Does it mean we want to have headers on AI Url or this check moved up/down a bit and is still performed? Is there a guarantee that request is not null?